src/Components/Counter/__tests__/Counter.test.js

```diff
@@ -0,0 +1,41 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { Counter } from '../Counter';
+
+describe('Counter tests', () => {
+  const initialValue = 5;
+
+  test('initial value is rendered correctly', async () => {
+    // Arrange
+    // Act
+    render(<Counter initialValue={initialValue} />);
+
+    // Assert
+    expect(await screen.findByTestId('counter-div')).toHaveTextContent(initialValue);
```

Comment:
`findByTestId` should be used only if other options are not valid (https://testing-library.com/docs/queries/bytestid).
Here something like this can be done:

```suggestion
    expect(screen.getByText(initialValue)).toBeInTheDocument();
```

Also, `.findBy...` should not be used here. 
If we are checking if some element exists on a page `.getBy...` should be used.